### ISSUE TITLE
setIntersect over singletons throws exception

### ISSUE THREAD

**Issue by arcfide** [arrayfire/arrayfire#1675]
The following code fails with an abort. The exception is some sort of memory access violation. This occurs regardless of the use of iota() or constant() to create the singleton array. Using an array containing at least two elements works fine. This happens with both the CPU and the OPENCL backend. 

```
#include <arrayfire.h>

using namespace af;

int main(int argc, char *argv[])
{
	array a1 = iota(dim4(1), dim4(1), s32);
	array a2 = iota(dim4(5), dim4(1), s32);
	
	array a3 = setIntersect(a1, a2);

	af_print(a3);

	return 0;
}
```

**Comment by pavanky** [comment]
This is being fixed by https://github.com/arrayfire/arrayfire/pull/1668

This is related to: https://github.com/arrayfire/arrayfire/issues/1660

**Comment by arcfide** [comment]
Is there an estimate of how soon these fixes are likely to be in a release? I'm asking to estimate whether I need to implement workarounds for this or whether I can wait until it is fixed here.

**Comment by pavanky** [comment]
@arcfide this will be released next week. If you are in a hurry, you can hotfixes-3.4.2 branch with the patch applied.

**Comment by shehzan10** [comment]
Fixed in hotfix-3.4.2

**Comment by shehzan10** [comment]
Oops. Wrong one.

**Comment by shehzan10** [comment]
Fixed in hotfix-3.4.2
